test/testdata/lsp/genericMethods.rb

```diff
@@ -15,24 +15,44 @@ def self.id(a)
     a
   end
 
+  sig {
+    type_parameters(:U)
+      .params(blk: T.proc.returns(T.type_parameter(:U)))
+      .returns(T.type_parameter(:U))
+  }
+  def self.block_id(&blk)
+    #      ^ def: block_id
+    yield
+  end
 end
 
 def main
   foo = Foo.new
   v1 = foo.id(1)
 # ^ hover: Integer
-         # ^ hover: sig { params(a: Integer).returns(Integer) }
-         # ^ usage: id
+  #        ^ hover: sig { params(a: T.type_parameter(:A)).returns(T.type_parameter(:A)) }
+  #        ^ usage: id
   v2 = foo.id("1")
 # ^ hover: String
-         # ^ hover: sig { params(a: String).returns(String) }
-         # ^ usage: id
+  #        ^ hover: sig { params(a: T.type_parameter(:A)).returns(T.type_parameter(:A)) }
+  #        ^ usage: id
   v3 = Foo.id(1)
 # ^ hover: Integer
-         # ^ hover: sig { params(a: Integer).returns(Integer) }
-         # ^ usage: staticid
+  #        ^ hover: sig { params(a: T.type_parameter(:A)).returns(T.type_parameter(:A)) }
+  #        ^ usage: staticid
   v4 = Foo.id("1")
 # ^ hover: String
-         # ^ hover: sig { params(a: String).returns(String) }
-         # ^ usage: staticid
+  #        ^ hover: sig { params(a: T.type_parameter(:A)).returns(T.type_parameter(:A)) }
+  #        ^ usage: staticid
+
+  v5 = Foo.block_id do
```

Comment:
Is it worth adding a hover for `v5` as well here?

Reply:
I don't think so. I think the result of the variable is already covered by other code paths.